Libraries/Lists/__tests__/VirtualizedList-test.js

```diff
@@ -506,161 +528,918 @@ describe('VirtualizedList', () => {
-    const stickyIndices = items
-      .filter(item => item.sticky)
-      .map(item => item.key);
+    const items = generateItemsStickyEveryN(10, 3);
 
     const ITEM_HEIGHT = 10;
 
     const component = ReactTestRenderer.create(
       <VirtualizedList
-        data={items}
-        stickyHeaderIndices={stickyIndices}
         initialNumToRender={5}
-        renderItem={({item}) => <item value={item.key} sticky={item.sticky} />}
-        getItem={(data, index) => data[index]}
-        getItemCount={data => data.length}
-        getItemLayout={(_, index) => ({
-          length: ITEM_HEIGHT,
-          offset: ITEM_HEIGHT * index,
-          index,
-        })}
+        {...baseItemProps(items)}
+        {...fixedHeightItemLayoutProps(ITEM_HEIGHT)}
       />,
     );
 
     expect(component).toMatchSnapshot();
   });
 
-  it('realizes sticky headers in viewport on batched render', () => {
-    const items = Array(10)
-      .fill()
-      .map((_, i) => (i % 3 === 0 ? {key: i, sticky: true} : {key: i}));
-    const stickyIndices = items
-      .filter(item => item.sticky)
-      .map(item => item.key);
-
+  it('renders sticky headers in viewport on batched render', () => {
+    const items = generateItemsStickyEveryN(10, 3);
     const ITEM_HEIGHT = 10;
 
-    const virtualizedListProps = {
-      data: items,
-      stickyHeaderIndices: stickyIndices,
-      initialNumToRender: 1,
-      windowSize: 1,
-      renderItem: ({item}) => <item value={item.key} sticky={item.sticky} />,
-      getItem: (data, index) => data[index],
-      getItemCount: data => data.length,
-      getItemLayout: (_, index) => ({
-        length: ITEM_HEIGHT,
-        offset: ITEM_HEIGHT * index,
-        index,
-      }),
-    };
-
     let component;
-
     ReactTestRenderer.act(() => {
       component = ReactTestRenderer.create(
-        <VirtualizedList key={'A'} {...virtualizedListProps} />,
+        <VirtualizedList
+          initialNumToRender={1}
+          windowSize={1}
+          {...baseItemProps(items)}
+          {...fixedHeightItemLayoutProps(ITEM_HEIGHT)}
+        />,
       );
     });
 
     ReactTestRenderer.act(() => {
-      component
-        .getInstance()
-        ._onLayout({nativeEvent: {layout: {width: 10, height: 50}}});
-      component.getInstance()._onContentSizeChange(10, 100);
-      jest.runAllTimers();
+      simulateLayout(component, {
+        viewport: {width: 10, height: 50},
+        content: {width: 10, height: 200},
+      });
+      performAllBatches();
     });
 
     expect(component).toMatchSnapshot();
   });
 
-  it('keeps sticky headers realized after scrolled out of viewport', () => {
-    const items = Array(20)
-      .fill()
-      .map((_, i) =>
-        i % 3 === 0 ? {key: i, sticky: true} : {key: i, sticky: false},
-      );
-    const stickyIndices = items
-      .filter(item => item.sticky)
-      .map(item => item.key);
-
+  it('keeps sticky headers above viewport visualized', () => {
```

Comment:
The prop (exposed to ScrollView) is [documented](https://reactnative.dev/docs/scrollview#stickyheaderindices), but I'm not sure virtualization behavior is. I discovered the logic in-code, but its requirement makes sense.

If you have a sticky header, it will stay at top of viewport, even if its original layout position is scrolled out of viewport. That means layout-based logic would not be aware it is on screen, and would consider it eligible to be virtualized away (while on screen). Exempting sticky headers (even just the closest) above the viewport from virtualization prevents them from disappearing.

Reply:
Oh yes that does make sense! Yea would be great to add this documentation to this test.